packages/field-base/src/input-constraints-mixin.js

```diff
@@ -70,7 +70,7 @@ export const InputConstraintsMixin = dedupingMixin(
       _createConstraintsObserver() {
         // This complex observer needs to be added dynamically instead of using `static get observers()`
         // to make it possible to tweak this behavior in classes that apply this mixin.
-        this._createMethodObserver(`_constraintsChanged(${this.constructor.constraints.join(', ')})`);
+        this._createMethodObserver(`_constraintsChanged(${this.constructor.constraints.join(', ')})`, true);
```

Comment:
**note:** `InputConstraintsMixin` had to be refactored because its constraints observer wasn't called initially for some components like `date-picker`, `email-field`, `number-field` when they had some constraints specified. This apparently has something to do with how Polymer handles observers and properties declared at different layers of inheritance. As a workaround, I added the second argument to `_createMethodObserver` that now guarantees that the observer is called regardless of constraints which also required refactoring  `_constraintsChanged` to make it only validate when the user specified any constraints.